mkt/websites/views.py

```diff
@@ -61,3 +70,200 @@ class ReviewersWebsiteSearchView(WebsiteSearchView):
                        SortingFilter]
     serializer_class = ReviewerESWebsiteSerializer
     form_class = ReviewersWebsiteSearchForm
+
+
+class WebsiteMetadataScraperView(CORSMixin, MarketplaceView, APIView):
+    """
+    Base website search view based on a single-string query.
+    """
+    cors_allowed_methods = ['get']
+    authentication_classes = [RestSharedSecretAuthentication,
+                              RestOAuthAuthentication]
+    permission_classes = [AllowAny]
+
+    errors = {
+        'no_url': '`url` querystring parameter required',
+        'network': 'Unable to fetch website metadata',
+        'malformed_data': 'Unable to parse website HTML.'
+    }
+
+    def get_name(self, doc):
```

Comment:
I'd suggest creating some helper methods like `get_title_text(doc)`, `get_meta_content(doc, selector)` and `get_link_href(doc, selector)` (these could take `doc(selector)` instead, too). If you can remove the `try`/`except` that would be better too, I'd say.

```py
def get_meta_content(el):
    el = doc(selector)
    if el and 'content' in el.attrib:
        return el.attrib['content']

def get_description(self, doc):
    return (get_meta_content('meta[name="description"]') or
            get_meta_content('meta[property="og:description"]') or
            get_meta_content('meta[name="msapplication-tooltip"]'))